src/layers/ArrayBitmapLayer.js

```diff
@@ -0,0 +1,107 @@
+import { BitmapLayer } from '@deck.gl/layers';
+import { Texture2D, Model, Geometry } from '@luma.gl/core';
+import GL from '@luma.gl/constants';
+
+import { PHOTOMETRIC_INTERPRETATIONS } from '../constants';
+
+const DEFAULT_TEXTURE_PARAMETERS = {
+  [GL.TEXTURE_MIN_FILTER]: GL.LINEAR,
+  [GL.TEXTURE_MAG_FILTER]: GL.LINEAR,
+  [GL.TEXTURE_WRAP_S]: GL.CLAMP_TO_EDGE,
+  [GL.TEXTURE_WRAP_T]: GL.CLAMP_TO_EDGE
+};
+
+export default class ArrayBitmapLayer extends BitmapLayer {
+  _getModel(gl) {
+    const {
+      channelData: { photometricInterpretation }
+    } = this.props;
+    let photometricInterpretationShader;
+    // This is a port to the GPU of a subset of https://github.com/geotiffjs/geotiff.js/blob/master/src/rgb.js
+    // Safari was too slow doing this off of the GPU and it is noticably faster on other browsers as well.
```

Comment:
Without imagery to test against, I don't feel comfortable implementing the rest.  I'm fine with a "take it as they come" approach to this.